lib/active_model/serializer.rb

```diff
@@ -14,40 +14,14 @@ class Serializer
     include Configuration
     include Associations
     require 'active_model/serializer/adapter'
-
-    # Matches
-    #  "c:/git/emberjs/ember-crm-backend/app/serializers/lead_serializer.rb:1:in `<top (required)>'"
-    #  AND
-    #  "/c/git/emberjs/ember-crm-backend/app/serializers/lead_serializer.rb:1:in `<top (required)>'"
-    #  AS
-    #  c/git/emberjs/ember-crm-backend/app/serializers/lead_serializer.rb
-    CALLER_FILE = /
-      \A       # start of string
-      \S+      # one or more non-spaces
-      (?=      # stop previous match when
-        :\d+     # a colon is followed by one or more digits
-        :in      # followed by a colon followed by in
-      )
-    /x
-
-    # Hashes contents of file for +_cache_digest+
-    def self.digest_caller_file(caller_line)
-      serializer_file_path = caller_line[CALLER_FILE]
-      serializer_file_contents = IO.read(serializer_file_path)
-      Digest::MD5.hexdigest(serializer_file_contents)
-    rescue TypeError, Errno::ENOENT
-      warn <<-EOF.strip_heredoc
-       Cannot digest non-existent file: '#{caller_line}'.
-       Please set `::_cache_digest` of the serializer
-       if you'd like to cache it.
-       EOF
-      ''.freeze
+    Attribute = Struct.new(:callable) do
+      delegate :call, to: :callable
     end
 
     with_options instance_writer: false, instance_reader: false do |serializer|
       class_attribute :_type, instance_reader: true
-      class_attribute :_attributes               # @api private : names of attribute methods, @see Serializer#attribute
-      self._attributes ||= []
+      class_attribute :serialized_attributes, instance_writer: false # @api public: maps attribute name to 'Attribute' instance
+      self.serialized_attributes ||= {}
       class_attribute :_attributes_keys          # @api private : maps attribute value to explict key name, @see Serializer#attribute
```

Comment:
Yeah, it is used for caching (although I think it could/should be avoided, but let's worry about that in an other PR).